### PR TITLE
Add tests for font measurement

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Move font quality method to extensions.
Add basic text measurement related tests.

Put the tests in a general file as the intent is to refactor these methods (e.g. they won't stay where they are).

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3538)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/SystemDrawingExtensions.cs [hunk 1]
```diff
@@ -11,5 +12,32 @@ internal static class SystemDrawingExtensions
     {
         internal static Gdi32.HBITMAP GetHBITMAP(this Bitmap bitmap) => (Gdi32.HBITMAP)bitmap.GetHbitmap();
         internal static Gdi32.HFONT ToHFONT(this Font font) => (Gdi32.HFONT)font.ToHfont();
+
+        /// <summary>
+        ///  Attempts to match the TextRenderingHint of the specified Graphics object with a LOGFONT.lfQuality value.
+        /// </summary>
+        internal static Gdi32.QUALITY FontQualityFromTextRenderingHint(this IDeviceContext? deviceContext)
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/System/Windows/Forms/SystemDrawingExtensions.cs:19]: Any reason not to move it to `TextRenderer` where it is used?

File: src/System.Windows.Forms/tests/UnitTests/Text/FontMetrics.cs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            Assert.Equal(expected, graphics.GetOverhangPadding(windowsFont));
+        }
+
+        [Theory(),
+            InlineData("Arial", 9.0f, 3, 4),
+            InlineData("Arial", 12.0f, 3, 5),
+            InlineData("Microsoft Sans Serif", 16.0f, 5, 7),
+            InlineData("Times New Roman", 11.0f, 3, 5),
+            InlineData("MS Gothic", 10.0f, 3, 4),
+            ]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/Text/FontMetrics.cs:78]: We don't combine the attributes
```suggestion
        [Theory]
        [InlineData("Arial", 9.0f, 3, 4)]
        [InlineData("Arial", 12.0f, 3, 5)]
        [InlineData("Microsoft Sans Serif", 16.0f, 5, 7)]
        [InlineData("Times New Roman", 11.0f, 3, 5)]
        [InlineData("MS Gothic", 10.0f, 3, 4)]
```

File: src/System.Windows.Forms/tests/UnitTests/Text/FontMetrics.cs [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+            Assert.Equal(right, margins.iRightMargin);
+        }
+
+        [Theory(),
+            InlineData("Arial", 9.0f, 73, 15),
+            InlineData("Arial", 12.0f, 95, 18),
+            InlineData("Microsoft Sans Serif", 16.0f, 136, 26),
+            InlineData("Times New Roman", 11.0f, 84, 17),
+            InlineData("MS Gothic", 10.0f, 91, 14),
+            ]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/Text/FontMetrics.cs:101]: ditto

File: src/System.Windows.Forms/tests/UnitTests/Text/FontMetrics.cs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+using System.Drawing;
+using System.Windows.Forms.Internal;
+using Xunit;
+using static Interop;
+
+namespace System.Windows.Forms.Tests.Text
+{
+    public class FontMetrics
+    {
+        [Theory()]
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/Text/FontMetrics.cs:14]: [nit] parens not needed
ditto below

File: src/System.Windows.Forms.Primitives/src/System/Windows/Forms/SystemDrawingExtensions.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms.Primitives/src/System/Windows/Forms/SystemDrawingExtensions.cs:19]: Nope, just missed the fact.

**Comment by RussKie** [comment]
Multiple tests fail
```
    System.Windows.Forms.Tests.CommonDialogTests.ShowDialog_ControlOwnerWithHandleWithVisualStyles_ReturnsExpected [SKIP]
      Crash with AbandonedMutexException. See: https://github.com/dotnet/arcade/issues/5325
    System.Windows.Forms.Tests.Text.FontMetrics.Font_GetOverhangPadding(family: "Arial", size: 9, expected: 2.5) [FAIL]
      System.NullReferenceException : Object reference not set to an instance of an object.
      Stack Trace:
        /_/src/System.Windows.Forms/tests/UnitTests/Text/FontMetrics.cs(69,0): at System.Windows.Forms.Tests.Text.FontMetrics.Font_GetOverhangPadding(String family, Single size, Single expected)
```

**Comment by JeremyKuhne** [comment]
> Multiple tests fail

Updated. Side effect of multiple threads running and not using the property.

File: src/System.Windows.Forms/src/System/Windows/Forms/TextRenderer.cs [hunk 5]
```diff
@@ -166,14 +156,17 @@ public static Size MeasureText(IDeviceContext dc, string? text, Font? font)
             {
```
Comment (hughbe) [inline src/System.Windows.Forms/src/System/Windows/Forms/TextRenderer.cs:156]: should we simplify the code for `MeasureText` into a new `MeasureTextInternal` as we did for `DrawText`

Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/TextRenderer.cs:156]: Yeah. I'll update shortly. I'm going to remove `WindowsGraphics` etc., this cleanup helps focus how we're actually using things.

**Pull request merged**